developer-extension/src/background/utils.ts

```diff
@@ -1,19 +1,19 @@
-export function evaluateCodeInActiveTab(code: () => void) {
+export function evaluateCodeInActiveTab(code: (arg?: string) => void, arg?: string) {
   chrome.tabs.query({ currentWindow: true, active: true }, (tabs) => {
     for (const tab of tabs) {
       if (tab.id) {
-        evaluateCodeInline(tab.id, code)
+        evaluateCodeInline(tab.id, code, arg)
       }
     }
   })
 }
 
-function evaluateCodeInline(tabId: number, code: () => void) {
+function evaluateCodeInline(tabId: number, code: (arg?: string) => void, arg?: string) {
   chrome.tabs.executeScript(tabId, {
     code: `{
       const script = document.createElement('script')
       script.setAttribute("type", "module")
-      script.textContent = ${JSON.stringify(`(${String(code)})()`)}
+      script.textContent = ${JSON.stringify(`(${String(code)})('${arg}')`)}
       document.body.appendChild(script)
       script.remove()
     }`,
```

Comment:
To avoid the contentscript and custom event passing, we could imagine using the `executeScript` return value to pass the result of the evaluated function. Something like that:

```ts
function evaluateCodeInline<R>(tabId: number, code: (arg?: string) => R, arg?: string): Promise<R> {
  return chrome.tabs.executeScript(tabId, {
    code: `{
      const script = document.createElement('script')
      script.setAttribute("type", "module")
      script.textContent = ${JSON.stringify(`window.DD_EXTENSION_RESULT = (${String(code)})(${JSON.stringify(arg)})`)}
      document.body.appendChild(script)
      script.remove();
      const result = window.DD_EXTENSION_RESULT;
      delete window.DD_EXTENSION_RESULT;
      result
    }`,
  })
}
```

Reply:
By testing your approach, it appears that window context is not shared between the page (where the added script tag is evaluated) and where the execute script code is evaluated. 
So the `DD_EXTENSION_RESULT` approach does not seem to work.
Did you have different results?

Reply:
Ah indeed. Maybe the event dispatch/listening could be done here then.

Reply:
probably, IMO this change would shift the complexity but not drastically reduce it like the first suggestion.
I'd be in favor of keeping the current implementation and revisit later.
wdyt?

Reply:
Sure, that's fine :)